src/server/pmix_server_ops.c

```diff
@@ -454,26 +450,21 @@ static pmix_server_trkr_t* new_tracker(char *id, pmix_proc_t *procs,
              * of the loop */
         }
         /* is this one of my local ranks? */
-        ns_local = 0;
         PMIX_LIST_FOREACH(info, &nptr->ranks, pmix_rank_info_t) {
             if (procs[i].rank == info->pname.rank ||
                 PMIX_RANK_WILDCARD == procs[i].rank) {
                     pmix_output_verbose(5, pmix_server_globals.base_output,
                                         "adding local proc %s.%d to tracker",
                                         info->pname.nspace, info->pname.rank);
                 /* track the count */
-                ns_local++;
+                trk->nlocal++;
```

Comment:
I don't think this will correctly handle the case where PMIX_RANK_WILDCARD is used for the procs[i].rank, but only some of those ranks are local.  In that case, it won't trigger line 437 and it willl go into this 'if' code.   Do we need to do a full count as in:

https://github.com/openpmix/openpmix/pull/1602
?

Reply:
I don't think either one is correct. Let me ponder this a bit more.

Reply:
Please see #1603  - I believe it resolves the problems and hopefully all the comments will help you to follow the code. Once we converge on it, I'll update this one.